arangod/Replication2/ReplicatedState/FollowerStateManager.tpp

```diff
@@ -156,19 +198,24 @@ void FollowerStateManager<S>::checkSnapshot(
 template<typename S>
 void FollowerStateManager<S>::startService(
     std::shared_ptr<IReplicatedFollowerState<S>> hiddenState) {
-  LOG_TOPIC("26c55", TRACE, Logger::REPLICATED_STATE)
-      << "starting service as follower";
-  state = hiddenState;
-  state->_stream = stream;
+  _guardedData.template doUnderLock([&](GuardedData& data) {
+    LOG_TOPIC("26c55", TRACE, Logger::REPLICATED_STATE)
+        << "starting service as follower";
+    data.state = hiddenState;
+  });
+  hiddenState->setStateManager(this->shared_from_this());
   pollNewEntries();
 }
 
 template<typename S>
 void FollowerStateManager<S>::ingestLogData() {
-  updateInternalState(FollowerInternalState::kTransferSnapshot);
-  auto demux = Demultiplexer::construct(logFollower);
-  demux->listen();
-  stream = demux->template getStreamById<1>();
+  auto core = _guardedData.template doUnderLock([&](GuardedData& data) {
```

Comment:
Removing the template keyword from here caused no problems on my machine, and might make MSVC happy. 
```suggestion
  auto core = _guardedData. doUnderLock([&](GuardedData& data) {
```